### ISSUE TITLE
Non-normalized eigenvectors in linalg.eigh for complex hermitian matrix on osx-arm64

### ISSUE THREAD

**Issue by allegro0132** [google/jax#12156]
### Description

Hi, the `jax.numpy.linalg.eigh` function return non-normalized eigenvectors, the bug only on Apple silicon SoC.

It's also appears in numpy on osx-arm64, but it fixed right now.  [Related link](https://lightrun.com/answers/numpy-numpy-bug-non-normalized-eigenvectors-in-linalgeigh-for-complex-hermitian-matrix-on-osx-arm64)

## The code example
```python
import numpy as np
import jax.numpy as jnp

mat = np.array([[
                    -2.73715202e-13 - 8.11929884e-23j,
                    1.27827591e+01 + 1.77027777e+01j,
                    -3.48766901e-13 - 4.19572569e-13j
                ],
                [
                    1.27827591e+01 - 1.77027777e+01j,
                    -1.69169696e-12 + 7.23263452e-16j,
                    3.18928587e+01 - 4.41682087e+01j
                ],
                [
                    -3.48756736e-13 + 4.19572569e-13j,
                    3.18928587e+01 + 4.41682087e+01j,
                    -2.05401813e-12 + 7.89088376e-24j
                ]])
eig_val_np, eig_vec_np = np.linalg.eigh(mat)
eig_val_jax, eig_vec_jax = jnp.linalg.eigh(mat)
```

## Error result
```python
eig_vec_jax @ jnp.conj(eig_vec_jax).T = 
array([
[ 1.00000000e+00+0.00000000e+00j, -7.09279869e-17-3.83554354e-17j, 3.29232499e-16+1.67077176e-22j],
[-7.09279869e-17+3.83554354e-17j, 1.00000000e+00-7.56437490e-18j, -1.86184468e-16+1.24903555e-16j],
[ 3.29232499e-16-1.67077176e-22j, -1.86184468e-16-1.12629232e-16j, 1.00000000e+00+5.04444916e-24j]
])
```

## Expected result
```python
eig_vec_np @ np.conj(eig_vec_np).T = 
DeviceArray([
[ 2.0207963 +0.j        , -0.02424392+0.03224995j, -0.14756264+0.17489427j],
[-0.02424392-0.03224995j,  1.9354674 +0.j        , 0.06381008+0.550426j  ],
[-0.14756264-0.17489427j,  0.06381008-0.550426j  , 1.5762056 +0.j        ]
], dtype=complex64)
```

### What jax/jaxlib version are you using?

jax v0.3.16, jaxlib v0.3.15

### Which accelerator(s) are you using?

CPU

### Additional System Info

macos
